### PR TITLE
Bugfix + support for new commands

### PR DESCRIPTION
**Pull request by DanielSmedegaardBuus**
As described in the commits :)

### CODE REVIEW

**Comment by DanielSmedegaardBuus** [comment]
Might wanna wait a bit ... It seems to not play nice with phantomjs :(

File: JavascriptFormatter.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ function filterForRemoteControl(originalCommands) {
             var c1 = c.createCopy();
             c1.command = c.command.replace(/AndWait$/, '');
             commands.push(c1);
-            commands.push(new Command("waitForPageToLoad", options['global.timeout'] || "30000"));
+            commands.push(new Command("waitForPageToLoad", "options.timeout || " + (options['global.timeout'] || "30000")));
```
Comment (flyingfisher) [inline JavascriptFormatter.js:47]:
```
  options['global.timeout']
```

This is wired.

File: JavascriptFormatter.ts [hunk 2]
```diff
@@ -1050,7 +1072,8 @@ function waitFor(expression) {
     return "waitFor(browser, function(browser){\n"
         + (expression.setup ? indents(1) + expression.setup() + "\n" : "")
         + indents(1) + "return " + expression.toString() + ";\n"
-        + indents(0) + "}, 30000);\n";
+        + indents(0) + "}, '" + expression.toString().replace("'", "\\'")
```
Comment (flyingfisher) [inline JavascriptFormatter.ts]:
```
  expression.toString().replace("'", "\\'")
```

This is only replace the first '.

**Comment by flyingfisher** [comment]
This change is a little complex. I think it should be wrapped in some test. Mocha or something like it.

File: JavascriptFormatter.js (hunk 1 above)
Comment (DanielSmedegaardBuus) [inline JavascriptFormatter.js:47]: Hi :)

Not sure what you mean. `options['global.timeout']` isn't defined anywhere that I can see, so I assumed it was something from your other project, like the "lbParm" from before?

Or do you mean that I shouldn't allow it to be changed at run-time?

Cheers :)

File: JavascriptFormatter.ts (hunk 2 above)
Comment (DanielSmedegaardBuus) [inline JavascriptFormatter.ts]: Woops :D

**Comment by DanielSmedegaardBuus** [comment]
I totally agree about the tests. But seeing as there aren't any tests at present, I'm not really prepared to start from scratch... Do you have any tests that aren't included with the repository? If so, it'd be great to be able to use them and extend from there :+1: 

The thing about phantomjs has nothing to do with these changes by the way, I'm apparently having some issues with my Selenium setup, so once I fix the mistakes, feel free to pull if you'd like :)

File: JavascriptFormatter.js (hunk 1 above)
Comment (flyingfisher) [inline JavascriptFormatter.js:47]:
```
  options['global.timeout'] !== options.global.timeout
```

This is what I mean.

**Comment by flyingfisher** [comment]
That should be my work. I will find a way to integrate test into project. Thanks

**Comment by flyingfisher** [comment]
I will merge this. And write some test for it later.

**Comment by DanielSmedegaardBuus** [comment]
Cool :+1:

File: JavascriptFormatter.js (hunk 1 above)
Comment (DanielSmedegaardBuus) [inline JavascriptFormatter.js:47]: Still don't know what you mean. I didn't put in options['global.timeout']; it was already there? I was just assuming you needed it for something even though I couldn't see where it came from, so I left it in? :)

Cheers

**Pull request merged**